controllers/snapshot/snapshot_adapter.go

```diff
@@ -0,0 +1,444 @@
+				if tag == "ephemeral" {
+					isEphemeral = true
+					break
+				}
+			}
+			if !isEphemeral {
+				availableEnvironments = append(availableEnvironments, environment)
+			}
+		}
+	}
+	return &availableEnvironments, nil
+}
+
+// findExistingApplicationSnapshotEnvironmentBinding attempts to find an ApplicationSnapshotEnvironmentBinding that's
+// associated with the provided environmentName
+func (a *Adapter) findExistingApplicationSnapshotEnvironmentBinding(environmentName string) (*appstudioshared.ApplicationSnapshotEnvironmentBinding, error) {
+	applicationSnapshotEnvironmentBindingList := &appstudioshared.ApplicationSnapshotEnvironmentBindingList{}
+	opts := []client.ListOption{
+		client.InNamespace(a.application.Namespace),
+		client.MatchingFields{"spec.environment": environmentName},
+	}
+
+	err := a.client.List(a.context, applicationSnapshotEnvironmentBindingList, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, binding := range applicationSnapshotEnvironmentBindingList.Items {
+		if binding.Spec.Application == a.application.Name {
+			return &binding, nil
+		}
+	}
+
+	return nil, nil
+}
+
+// getAllApplicationComponents loads from the cluster all Components associated with the given Application.
+// If the Application doesn't have any Components or this is not found in the cluster, an error will be returned.
+func (a *Adapter) getAllApplicationComponents(application *hasv1alpha1.Application) (*[]hasv1alpha1.Component, error) {
+	applicationComponents := &hasv1alpha1.ComponentList{}
+	opts := []client.ListOption{
+		client.InNamespace(application.Namespace),
+		client.MatchingFields{"spec.application": application.Name},
+	}
+
+	err := a.client.List(a.context, applicationComponents, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &applicationComponents.Items, nil
+}
+
+// markSnapshotAsPassed updates the result label for the ApplicationSnapshot
+// If the update command fails, an error will be returned
+func (a *Adapter) markSnapshotAsPassed(applicationSnapshot *appstudioshared.ApplicationSnapshot, message string) (*appstudioshared.ApplicationSnapshot, error) {
+	patch := client.MergeFrom(applicationSnapshot.DeepCopy())
+	meta.SetStatusCondition(&applicationSnapshot.Status.Conditions, metav1.Condition{
+		Type:    "HACBSTestSucceeded",
+		Status:  metav1.ConditionTrue,
+		Reason:  "Passed",
+		Message: message,
+	})
+	err := a.client.Status().Patch(a.context, applicationSnapshot, patch)
+	//err := a.client.Status().Update(a.context, applicationSnapshot)
+	if err != nil {
+		return nil, err
+	}
+	return applicationSnapshot, nil
+}
+
+// isSnapshotConditionMet returns a boolean indicating whether the Snapshot's status has the conditionType set to true
+func (a *Adapter) isSnapshotConditionMet(snapshot *appstudioshared.ApplicationSnapshot, conditionType string) bool {
+	condition := meta.FindStatusCondition(snapshot.Status.Conditions, conditionType)
+	if condition != nil {
```

Comment:
Replace everything with `return condition != nil && condition.Status != metav1.ConditionUnknown`

Reply:
Ack, changed it to that